dotcom-rendering/src/lib/useCommentCount.ts

```diff
@@ -0,0 +1,33 @@
+import { isNonNullable } from '@guardian/libs';
+import { isServer } from './isServer';
+import { useApi } from './useApi';
+
+type CommentCounts = Record<string, number>;
+
+const getInitialSetFromDOMAttribute = (attribute: string) =>
+	[...document.querySelectorAll(`[${attribute}]`)]
+		.map((element) => element.getAttribute(attribute))
+		.filter(isNonNullable);
```

Comment:
I assume we are pulling these elements from the DOM because they are side loaded and not part of the React application?

Reply:
We need to pass the initial set of all emements on the page, but we do not want to pass the whole set to each and every `CommentCount` island… so this is leveraging the existing convention of adding `data-discussion-id` attributes. There may be a better way of doing this.

Reply:
Non-blocking. I will catch up with you to understand this feature better.